docs/guides/08-interactivity-events.md

```diff
@@ -0,0 +1,411 @@
+        frameBorder="0">
+    </iframe>
+</div>
+
+#### Data-driven variables
+If the data you are interested in for your interaction is a feature `property`, such as the _name_ of the city or its _population_, then you can also use some variables to store them. Those are called _data-driven variables_, cause their values change as you interact with each of the features (in our example, with each city).
+
+To test them you should edit again your viz, so it looks like this:
+```js
+const viz = new carto.Viz(`
+    @currentFeatures: viewportFeatures()
+    @name: $name
+    @popK: $pop_max / 1000.0
+`);
+```
+> Both properties, $name and $pop_max are columns in the original dataset.
+
+As the variables depend on properties, you can't just access them using something like `viz.variables.name.value` on `layer:updated`. That will throw an error saying: _property needs to be evaluated in a 'feature'_. You need to use [carto.Interactivity](/developers/carto-vl/reference/#cartointeractivity).
+
+
+### Feature events
+All feature interactions are ruled by the `carto.Interactivity`, so let's create an object of this type, associating it with the current layer.
+
+Add this line:
+```js
+const interactivity = new carto.Interactivity(layer);
+```
+
+Then use its `featureClick` event to react when you click on a city:
+```js
+interactivity.on('featureClick', featureEvent => {
+    featureEvent.features.forEach((feature) => {
+        const name = feature.variables.name.value;
+        const popK = feature.variables.popK.value.toFixed(0);
+        console.log(`You have clicked on ${name} with a population of ${popK}K`);
+    });
+});
+```
+> Notice how `carto.Interactivity` provides you for free with a dynamic change on the mouse pointer when you hover on a feature, and how it handles a collection, cause you can click on several features at the same time if they are near enough.
+
+The `carto.Interactivity` can handle different events:
+- `featureClick`: Fired when the user clicks on features.
+- `featureClickOut`: Fired when the user clicks outside a feature that was clicked in the last featureClick event.
+- `featureHover`: Fired when the user moves the cursor over a feature.
+- `featureEnter`: Fired the first time the user moves the cursor inside a feature.
+- `featureLeave`: Fired the first time the user moves the cursor outside a feature.
+
+In every callback a single parameter of type [featureEvent](/developers/carto-vl/reference/#featureevent) will be received. This object will have the `position` and `coordinates` where the
+event happened (we didn't use that so far) and the list of [Features](/developers/carto-vl/reference/#feature) that have been interacted.
+
+
+### Adding pop-ups
+A very common case is to display pop-ups, little emerging windows with information on the features.
+
+You can build the pop-up yourself if you want, but using `Mapbox GL` allows you to easily reuse [mapboxgl.Popup](https://www.mapbox.com/mapbox-gl-js/api/#popup).
```

Comment:
You can build the pop-up yourself if you want --> You can build the pop-up yourself if you want to